src/hooks/applications/useApplications.ts

```diff
@@ -0,0 +1,81 @@
+import { useDataQuery } from "@/hooks/useDataQuery";
+import { Dispatch, SetStateAction, useEffect, useMemo, useState } from "react";
+import { keepPreviousData, useQueryClient } from "@tanstack/react-query";
+import { PROJECTS_PAGE_SIZE } from "@common/consts";
+import { ICappResponse } from "@models/applications/applications";
+
+interface IApplicationsData {
+  remainingCount: number;
+  count: number;
+  capps: ICappResponse[];
```

Comment:
Creating a model for http response can be confusing, also, it's only used here so i think it can be deleted.
```suggestion
  capps: {capp: ICappData}[];
```